File: backend/utils/twosum.java
import java.util.*;
public class Main {
    public static int[] twoSum(int[] nums, int target) {
        // Key: number, Value: index
        HashMap<Integer, Integer> numMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            // Check if complement exists
            if (numMap.containsKey(complement)) {
                return new int[]{numMap.get(complement), i};
            }
            // Store current number and index
            numMap.put(nums[i], i);
        }
        return new int[]{}; // no solution
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }
        int target = sc.nextInt();
        int[] result = twoSum(nums, target);
        if (result.length != 0) {
            System.out.print("[" + result[0] + "," + result[1] + "]");
        } else {
            System.out.print("[]");
        }
        sc.close();
    }
}

File: backend/utils/twoSum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;
vector<int> twoSum(vector<int>& nums, int target) {
    // Key: the number, Value: its index
    unordered_map<int, int> numMap;
    
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        
        // If complement exists in our map, we found the pair
        if (numMap.find(complement) != numMap.end()) {
            return {numMap[complement], i};
        }
        
        // Otherwise, add current number and its index to the map
        numMap[nums[i]] = i;
    }
    
    return {}; // Return empty if no solution is found
}
int main() {
    int target,n;
	cin>>n;
vector<int> nums(n);
for (int i = 0; i < n; i++) {
    cin >> nums[i];
}
	cin>>target;
    
    vector<int> result = twoSum(nums, target);
    
    if (!result.empty()) {
        cout << "[" << result[0] << "," << result[1] << "]";
    } else {
        cout << "[]";
    }
    
    return 0;
}